### ISSUE TITLE
2.1.3 Observable#concatMapEager queue size issue

### ISSUE THREAD

**Issue by Aschraf** [ReactiveX/RxJava#5608]
Hello,

I've got some issues with ConcatMapEager prefetch value. 

It seems that concatMapEager stops before the end of the Observable.

Here is a small example showing the issue : 

```java
Observable.range(1, 100000)
        .buffer(10)
        .concatMapEager(value -> Observable.just(value)
                .subscribeOn(Schedulers.io())
                .doOnNext(it -> Thread.sleep(ThreadLocalRandom.current().nextLong(10, 100))),
            6, 4)
        .blockingSubscribe(it -> System.out.println("Received : " + it));
```

This small code is supposed to write list of ints till 100000. It stops before finishing.

The code doesn't work for the default ConcatMapEager function either (prefetch = 128)

**Comment by akarnokd** [comment]
Thanks for reporting. Fixed in #5609.

**Comment by akarnokd** [comment]
Closing via #5609.
